### PR TITLE
Switch to dynamic-graphql

### PR DESCRIPTION
**Pull request by ricopinazo**

### CODE REVIEW

File: flux-capacitor/src/main.rs [hunk 1]
```diff
@@ -66,18 +59,19 @@ async fn main() {
...
+        .finish()
+        .unwrap();
+    let app = Route::new()
+        .at("/", get(graphql_playground).post(GraphQL::new(schema)))
+        .at("/health", get(health));
 
-    Server::bind(&"0.0.0.0:8000".parse().unwrap())
-        .serve(app.into_make_service())
-        .with_graceful_shutdown(shutdown_signal())
+    println!("Playground: http://localhost:8000");
```
Comment (miratepuffin) [inline flux-capacitor/src/main.rs:72]: Can we change to 1736

File: flux-capacitor/src/model/algorithm.rs [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    }
+}
+
+impl<G: GraphViewOps> Register for Algorithms<G> {
+    fn register(registry: Registry) -> Registry {
+        let mut registry = registry;
+        let mut object = Object::new("Algorithms");
+
+        object = object.field(Field::new(
+            "numVerticesAlgo",
```
Comment (miratepuffin) [inline flux-capacitor/src/model/algorithm.rs:30]: Can we move the example to a separate package.

File: flux-capacitor/src/model/algorithm.rs [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                let graph = &algos.graph;
+                FieldFuture::new(async move { Ok(Some(FieldValue::value(graph.num_vertices()))) })
+            },
+        ));
+
+        let algos = HashMap::from([("pagerank", Pagerank::register_algo::<G>)]);
+
+        for (name, register_algo) in algos {
+            (registry, object) = register_algo(name, registry, object);
+        }
```
Comment (miratepuffin) [inline flux-capacitor/src/model/algorithm.rs:43]: Make a ticket for publishing to docker where for graphql we can just copy over the plugin folder for a user.

Comment (miratepuffin) [reply flux-capacitor/src/model/algorithm.rs:43]: Check if the docker container still works with Poem

File: flux-capacitor/src/model/algorithm.rs [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+        // TODO: uncomment this for plugins
```
Comment (miratepuffin) [inline flux-capacitor/src/model/algorithm.rs:45]: Change this to point towards all files in a plugin directly.

**Review by miratepuffin — CHANGES_REQUESTED** [review]
Couple of issues for us to create, but otherwise LGTM

File: flux-capacitor/src/model/algorithm.rs [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    rank: f32,
+}
+
+impl From<(String, f32)> for Pagerank {
+    fn from((name, rank): (String, f32)) -> Self {
+        Self { name, rank }
+    }
+}
+
+impl Algo for Pagerank {
```
[the remark this answers is not in the document]
Comment (ricopinazo) [reply flux-capacitor/src/model/algorithm.rs:109]: Done

File: flux-capacitor/src/main.rs (hunk 1 above)
Comment (ricopinazo) [reply flux-capacitor/src/main.rs]: Done

File: flux-capacitor/src/model/algorithm.rs (hunk 2 above)
Comment (ricopinazo) [reply flux-capacitor/src/model/algorithm.rs]: removed it as we already have pagerank, was just for testing

**Pull request merged**